core/src/main/java/com/datastax/oss/driver/api/core/tracker/RequestTracker.java

```diff
@@ -71,6 +98,7 @@ default void onSuccess(
   default void onError(
       @NonNull Request request,
       @NonNull Throwable error,
+      // TODO: Shall we expose start and end timestamp so that users do not need to call nanoTime()?
```

Comment:
I'm guessing in the context of tracing it could be pretty useful to know what the start / end times are.

I think this change is probably invasive enough that we would want to do it in the next major, so we could consider doing this here.  This could also extend to all the other methods, in addition to removing any deprecated methods.